ext/DynamicHMCExt.jl

```diff
@@ -12,10 +26,15 @@ To use it, make sure you have DynamicHMC package (version >= 2) loaded:
 using DynamicHMC
 ```
 """ 
-struct DynamicNUTS{AD,space} <: Hamiltonian{AD} end
+struct DynamicNUTS{AD,space} <: Turing.Inference.Hamiltonian{AD} 
+    sampler::DynamicHMC.NUTS
+end
```

Comment:
This will cause type inference issues, can we use
```suggestion
struct DynamicNUTS{AD,space,T<:DynamicHMC.NUTS} <: Turing.Inference.Hamiltonian{AD} 
    sampler::T
end
```
instead?

Reply:
I implemented something like what you proposed but I feel like it is a little bit hacked together. 

```
struct DynamicNUTS{AD,space,T<:DynamicHMC.NUTS} <: Turing.Inference.Hamiltonian{AD} 
    sampler::T
end

DynamicNUTS(args...) = DynamicNUTS{Turing.ADBackend()}(args...)
DynamicNUTS{AD}(spl::DynamicHMC.NUTS, space::Tuple) where AD = DynamicNUTS{AD, space, DynamicHMC.NUTS}(spl)
DynamicNUTS{AD}(spl::DynamicHMC.NUTS) where AD = DynamicNUTS{AD}(spl, ())
DynamicNUTS{AD}() where AD = DynamicNUTS{AD}(DynamicHMC.NUTS())
Turing.externalsampler(spl::DynamicHMC.NUTS) = DynamicNUTS(spl)

DynamicPPL.getspace(::DynamicNUTS{<:Any, space}) where {space} = space
```

Any suggestions for improvements?

Reply:
sorted!